service/matching/task_queue_partition_manager.go

```diff
@@ -257,6 +259,14 @@ func (pm *taskQueuePartitionManagerImpl) AddTask(
 	return assignedBuildId, false, spoolQueue.SpoolTask(params.taskInfo)
 }
 
+func (pm *taskQueuePartitionManagerImpl) shouldBacklogSyncMatchTaskOnError(err error) bool {
+	if err != nil && errors.Is(err, consts.ErrResourceExhaustedBusyWorkflow) {
```

Comment:
I don't think this can work. `consts.ErrResourceExhaustedBusyWorkflow` doesn't have its `st` field initialized while that field will be set if the error is returned from rpc stack and will cause `errors.Is` to return false I think.